tool/src/org/antlr/v4/tool/GrammarParserInterpreter.java

```diff
@@ -64,7 +67,7 @@ public GrammarParserInterpreter(Grammar g, ATN atn, TokenStream input) {
 			  input);
 		this.g = g;
 		decisionStatesThatSetOuterAltNumInContext = findOuterMostDecisionStates();
-		stateToAltsMap = new int[g.atn.getNumberOfDecisions()][];
+		stateToAltsMap = new int[g.atn.states.size()][];
```

Comment:
looks like I can't use the decision number because (x)* for left recursive rules with only a single alternative do not get a decision number.